File: src/chicken/cli/commands/wall.clj
(ns chicken.cli.commands.wall
  (:require [chicken.repository :as rep]
            [chicken.cli.app :refer (pluginCommand timeFormat)]))

(def ^:private TOKEN " wall")

(defn validate [input]
  (if (not (.endsWith (.trim input) TOKEN))
    {:errors '("Invalid 'wall' format. Type 'Username wall'")}
    :ok))

(defn wall [input]
  "Print wall for username from input if it exists"
  (let
    [name (.trim (.substring input 0 (.indexOf input TOKEN)))
     wall (rep/getWall name)]
    (println)
    (doseq [post wall]
      (printf "%s - %s (%s)\n" (:author post) (:text post) (timeFormat (:timestamp post))))
    (println)))

(pluginCommand :token TOKEN :do wall :validate validate)

File: src/main/scala/chicken/cli/commands/Wall.scala
package chicken.cli.commands

import chicken.cli.App

case class Wall() extends CliCommand {

  def token = " wall"

  override def validate(input: String) = {
    if (!input.trim.endsWith(token))
      "Invalid 'wall' format. Type 'Username wall'" :: Nil
    else
      Nil
  }

  def exec(input: String) = {
    val name = input.substring(0, input.indexOf(token))
    println()
    App.rep.getWall(name).
      map(p => "%s - %s (%s)".format(p.author, p.text, App.timeFormat(p.timestamp))).
      foreach(println)
    println()
  }

}
